src/mon/FSCommands.cc

```diff
@@ -1525,6 +1513,22 @@ int FileSystemCommandHandler::_check_pool(
   }
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
+  auto app_map = pool->application_metadata;
+
+  if(!allow_overlay && !app_map.empty()) {
+    if(app_map.find(pg_pool_t::APPLICATION_NAME_CEPHFS) != app_map.end()) {
+      auto& [app_name, app_metadata] = *app_map.find(pg_pool_t::APPLICATION_NAME_CEPHFS);
+      auto& [pool_type, filesystem] = *app_metadata.begin();
+      *ss << "RADOS pool '" << pool_name << "' is already used by filesystem '"
```

Comment:
you should be checking for existance of `app_metadata["data"]` or `app_metadata["metadata"]` specifically instead of relying on `*app_metadata.begin()`

Reply:
Done